### ISSUE TITLE
Moving files to completly wrong folder

### ISSUE THREAD

**Issue by StreetDebugger** [Sonarr/Sonarr#4320]
Since the beta release of februarary 6th (that‘s my impression by now), i have the problem that some files are moved to completly wrong destination folders. It happens in about 1 of 50 files. The attached screenshot (sorry for the quality) shows the problem.

![AEDE5DBA-53DF-4719-9E1F-3FB7194D2D30](https://user-images.githubusercontent.com/78885902/107588269-a8f0ff00-6c03-11eb-8405-1a385ee59908.jpeg)

F: is a synology
B: is a rclone mounted drive, connected with box.com. Rclone is on the latest version.
Installation is running on Win10.
Sonarr is on 3.04.1126

It is possible, that the problem existed before, but i never saw it before. The first time i saw this to happen was after installing the latest Sonarr version on the 6th of februry.

Thanks for having a look at this!

**Comment by bakerboy448** [comment]
Please use one of the support channels: [forums](https://forums.sonarr.tv/), [subreddit](https://www.reddit.com/r/sonarr/), [discord ](https://discord.gg/Ex7FmFK), or [IRC ](http://webchat.freenode.net/?channels=#sonarr)for support/questions.

Also closing as well due to template not followed.

**Comment by StreetDebugger** [comment]
I‘m new to Github, sorry for making a mistake. I‘m not looking for support, i was reporting a bug i‘ve investigated for hours. Please re-open the issue. Thanks.

**Comment by StreetDebugger** [comment]
> Please use one of the support channels: [forums](https://forums.sonarr.tv/), [subreddit](https://www.reddit.com/r/sonarr/), [discord ](https://discord.gg/Ex7FmFK), or [IRC ](http://webchat.freenode.net/?channels=#sonarr)for support/questions.
> 
> Also closing as well due to template not followed.
> 
> There is absolutely no problem shown in that picture. it appears sonarr moved from your download folder on syno to the folder you have the series set to on box, as expected.

Please look at the screenshot again.

It found „the magic school bus“, but the source was „Just roll with it“. The „Just roll with it“ episode was imported to „The magic school bus“, which is wrong.

**Comment by bakerboy448** [comment]
Pretty sure it's user error...hence hop to one of the support channels to hash it out.

But regardless,
[Template was not followed; no logs provided](https://github.com/Sonarr/Sonarr/issues/new?assignees=&labels=&template=bug_report.md&title=); thus issue is staying closed.

**Comment by StreetDebugger** [comment]
> Pretty sure it's user error...hence hop to one of the support channels to hash it out.
> 
> But regardless,
> [Template was not followed; no logs provided](https://github.com/Sonarr/Sonarr/issues/new?assignees=&labels=&template=bug_report.md&title=); thus issue is staying closed.

I have no helpful informations in the logs as my system on log level „info“ at the time this happened. It took me 6 hours to find an example and make a screenshot of the details. Sorry, but i cannot provide more informations as i first would have to reproduce it with log level „debug“, which is set now.

And i cannot see where i could have made a user error here. From my perspective, the informations in the „name“ field do not match the „source“ field. If i‘d know if these informations are gathered by Sonarr by looking at the disk oder received from the download-client, i could investigate further.

**Comment by markus101** [comment]
This looks like Sonarr was told to import a file for that TMSB episode, but instead was given a path to a download for JRWI, which would indicate an issue with the download client pointing it at the wrong directory. The only way to fully investigate this would be with trace logs that were running before the import occurred (capturing the response from the download clients). If your download client still has those items in it's history you _may_ be able to use trace logs to see if a download is telling Sonarr to import from the wrong directory, either the wrong job folder or the parent folder. If you're unsure if what you're seeing in your logs is what I've described above then go through one of the support channels to clarify first.

Given there aren't people flooding to the support channels reporting this issue, it strongly indicates to us that this is a configuration issue, not a bug.

If you're able to reproduce again (not just trace logs of the current history, after importing) the issue _and_ it's not what I've explained above, edit your first post, following the bug template, including debug logs (real screenshots, aka not pictures of your screen, try Win+SHift+S) and reply letting us know it's been updated.

**Comment by StreetDebugger** [comment]
Thanks, i will investigate this further. Sorry for the screen picture, i work from remote and had no quick way of doing it better.

**Comment by StreetDebugger** [comment]
My investigations resulted in these points:

- Sonarr seems to get confused when there are *.mkv files in die download folder itself. That seem to have caused some trouble here. Not sure if this is an issue, just wat i saw.
- Importing obfuscated files seems to find episode numbers from the filenames. I‘ve added a cleanup script which runs on the download folder and removes all numbers from to-be-imported filenames. Same with - an . in filenames. Again, not sure if this is really a problem, in my constellation, it was one.
- Having sub-folders in the episode folder seems to complicate the process, especially when there are numers in the folder name. I‘ve solved it by doing unrar e in the download client. With that, there will no subfolders at all.

My test case was the download of „JAG“ in german as this series has obfuscated filenames in sub-folders and some of them even have samples. 

Basically my cleanup script does the following:
- remove 0-9, „-„, „.“ and „ „ from filenames
- Identify the file‘s content (AVI, Matroska etc.) and the add a respective extension

Import works like a charm now. Not reporting an issue, but probably these informations could be of some help for anyone else.

**Comment by bakerboy448** [comment]
Obfuscated files should be ignored completely in v3.  If there’s a new obfuscation method, well we’re playing wack-a-mole.  V2 is seeing no further development.

If that script runs on non-obfuscated files it may cause some issues.

without trace logs as requested by the template and as requested by both Markus and I, can’t tell you what’s going on with the .mkv issue....but if it’s an issue with v2... we’ll see if it’s a problem on v3.

**Comment by StreetDebugger** [comment]
> Obfuscated files should be ignored completely in v3. If there’s a new obfuscation method, well we’re playing wack-a-mole. V2 is seeing no further development.

I know, i‘m on 3.04.1126.

> If that script runs on non-obfuscated files it may cause some issues.

My script only handles obfuscated files and renames them as correct as possible.

For example: 15.09.2014.1 will be changes to something like ABCDEFGHI.avi (if the content is in fact an avi). There is a check in place that other files won‘t get overwritten.

> without trace logs as requested by the template and as requested by both Markus and I, can’t tell you what’s going on with the .mkv issue....but if it’s an issue with v2... well see if it’s a problem on v3.

Thanks for your feedbacks. The reason why i opened this case was the problem with Sonarr moving files to complete wrong folders. As i know now, having *.mkv files in the download folder directly seems to have caused that mess. I made sure, this will not happen again, it was caused by the VideoSort Script, which was working in nzbget. I‘ve turned that of and wrote my own script to rename obfuscated filenames within the correct episode folder. Everything is in the correct place right now.

My problem is solved, so you don‘t have to look at it any further. If you‘re still want to look at this, i can reproduce the problem and provide logs ... but i don‘t want to waste your time as the source of the problem was in fact data laying around in the wrong place. Let le know if it‘s of interest for you to look at this. For me, everything‘s fine now. Thanks!

**Comment by bakerboy448** [comment]
Makes sense as far as the cause and sounds like more or less what Markus expected

for the obfuscated file, can likely get a fix in by this weekend if not today.

just need to know the parent folder and the obfuscated file name.  Like I said it’s Wack-a-mole, but we can’t fix what we don’t know.

some.release.name/somegarbledfile.ext

**Comment by StreetDebugger** [comment]
Folder name: JAG.S04E24.German.FS.DVDRIP.XviD-AMiTOUS
Obfuscated folder within: gtjrtds3zjk7hgre34ghf ... or any garbage like that, does not matter.
File within the obfuscated folder: 15.09.2014.1

The 15.09.2014.1 is an AVI.

If you can, get some files from the TV show „JAG“ in german, Season 01. They have a lot of such garbage in them. If you need files, i can get them for you, but only for debugging reasons. You know, legal issues with passing media along...

**Comment by bakerboy448** [comment]
I literally need the exact path as I asked.

if you had provided logs as asked those would have it too.

I don’t need /path/to/downloads/thisfolder/garbageobfuscation 

but I do need /thisfolder/garbageobfuscation exactly as it downloaded. No edits. No partial info. The exact path as I asked.

**Comment by StreetDebugger** [comment]
Folder: JAG.S05E06.Bomben.auf.New.York.German.WS.DVDRip.Xvid.iNSPiRES
Subfolder within: 7701a2cd_0fae7b7d_b6b8cda8_5ed6240
Files within that subfolder: 
- 15.09.2014.1
- 4ever.html
- A_Linux_2rename.sh
- Click.rar
- MUST_READ.txt
- Thats_the_Board.nfo
- UnRAR.exe
- What.rar
- Win_click2rename.bat

I‘ve saved that thing here, so will be able to send it to you if needed.

**Comment by StreetDebugger** [comment]
And the download path is: f:/Download-nzbget/done/Series. The same as in my previous screenshot.

**Comment by bakerboy448** [comment]
Don’t think there’s anything to do about that one actually. There’s not even a valid video file in that hot garbage mess of a release.

yeah you’ll be stuck with your script for those.  Even if we handle those being obfuscated there is no video file to import.


For the one that was in your initial post, that type of obfuscation should be handled for sure

**Comment by StreetDebugger** [comment]
> Don’t think there’s anything to do about that one actually. There’s not even a valid video file in that hot garbage mess of a release.

There is a video file, but i cannot be identified by its file extension. My workaround-script reads the file, „analyzes“ the content ant set a correct file extension. That‘s quite easy code, no rocket science.

.avi = RIFF at the beginning
.mkv = „matroska“ in the first 100 bytes of the file
a.s.o.

> yeah you’ll be stuck with your script for those. Even if we handle those being obfuscated there is no video file to import.

I‘m fine with my workaround-script.

Thanks & have a nice day!
